packages/@aws-cdk/aws-apigatewayv2/lib/http/api-definition.ts

```diff
@@ -0,0 +1,181 @@
+import * as s3 from '@aws-cdk/aws-s3';
+import * as s3_assets from '@aws-cdk/aws-s3-assets';
+import * as cxapi from '@aws-cdk/cx-api';
+import { Construct, Node } from 'constructs';
+import { CfnApi } from '../apigatewayv2.generated';
+import { IHttpApi } from './api';
+
+/**
+ * Represents an OpenAPI definition asset.
+ */
+export abstract class ApiDefinition {
+  /**
+   * Creates an API definition from a specification file in an S3 bucket
+   */
+  public static fromBucket(bucket: s3.IBucket, key: string, objectVersion?: string): S3ApiDefinition {
+    return new S3ApiDefinition(bucket, key, objectVersion);
+  }
+
+  /**
+   * Create an API definition from an inline object. The inline object must follow the
+   * schema of OpenAPI 2.0 or OpenAPI 3.0
+   */
+  public static fromInline(definition: any): InlineApiDefinition {
+    return new InlineApiDefinition(definition);
+  }
+
+  /**
+   * Loads the API specification from a local disk asset.
+   */
+  public static fromAsset(file: string, options?: s3_assets.AssetOptions): AssetApiDefinition {
+    return new AssetApiDefinition(file, options);
+  }
+
+  /**
+   * Called when the specification is initialized to allow this object to bind
+   * to the stack, add resources and have fun.
+   *
+   * @param scope The binding scope. Don't be smart about trying to down-cast or
+   * assume it's initialized. You may just use it as a construct scope.
+   */
+  public abstract bind(scope: Construct): ApiDefinitionConfig;
```

Comment:
Based off the context I'm seeing here, it looks like this is a function the end user isn't meant to use. Given that, let's mark it internal and name it `_bind` instead.